### ISSUE TITLE
"error: ‘realpath’ was not declared in this scope" on cygwin64

### ISSUE THREAD

**Issue by TheElementalOfCreation** [avast/yaramod#25]
```
[ 83%] Building CXX object src/CMakeFiles/yaramod.dir/utils/filesystem.cpp.o
/cygdrive/c/Users/Creation/Downloads/git/retdec/external/src/yaramod-project/src/utils/filesystem.cpp: In function ‘std::string yaramod::detail::absolutePath(const string&)’:
/cygdrive/c/Users/Creation/Downloads/git/retdec/external/src/yaramod-project/src/utils/filesystem.cpp:86:6: error: ‘realpath’ was not declared in this scope
  if (realpath(path.c_str(), absolutePathStr) == nullptr)
      ^~~~~~~~
/cygdrive/c/Users/Creation/Downloads/git/retdec/external/src/yaramod-project/src/utils/filesystem.cpp:86:6: note: suggested alternative: ‘path’
  if (realpath(path.c_str(), absolutePathStr) == nullptr)
      ^~~~~~~~
      path
make[2]: *** [src/CMakeFiles/yaramod.dir/build.make:373: src/CMakeFiles/yaramod.dir/utils/filesystem.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:109: src/CMakeFiles/yaramod.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

**Comment by s3rvac** [comment]
Hi. I do not have Cygwin at hand at the moment. Would it be possible for you to check whether the following approach fixes the issue for you?
```
cd yaramod
sed -i 's/CMAKE_CXX_EXTENSIONS OFF/CMAKE_CXX_EXTENSIONS ON/' CMakeLists.txt
rm -rf build
mkdir build
cd build
cmake ..
make
```
The premise is that `realpath()` is not defined in Cygwin when compiling with `-std=c++14` instead of `-std=gnu++14` (this is what setting `CMAKE_CXX_EXTENSIONS` to `ON` does).

**Comment by metthal** [comment]
Should be fixed. Thanks @s3rvac for idea.

**Comment by TheElementalOfCreation** [comment]
From the `sed` command I get `Invalid option "-i"`

**Comment by TheElementalOfCreation** [comment]
Oh wait, ignore that. Bad copy-paste put `ed` instead of `sed`

**Comment by TheElementalOfCreation** [comment]
Also trying to make it from `build` returns an error for me because the makefile was put in the above directory

**Comment by s3rvac** [comment]
@TheElementalOfCreation I am not sure if I understand. Could you please try to build the current `master` (without any modifications)? @metthal has fixed this in there (commit 428f6b5e7d5de16389dcdc56d6f4f1361761cde0).
